network/src/peer_manager/mod.rs

```diff
@@ -533,6 +539,13 @@ where
                     );
                     self.send_conn_notification(peer_id, notif);
                 }
+
+                // Garbage collect unused rate limit buckets
```

Comment:
this implicitly makes an assumption that we have some sort of rate limiting on incoming connections ... otherwise the attacker is incentivized to just create a new connection to drive in more traffic... maybe we should start buckets out as empty or 1/4 full?

Reply:
So, originally what I wanted to do was have the buckets garbage collected on a periodic basis, ignoring when connections are created and deleted, which made more sense to be full.

It is a good idea to start empty, except it does slow *all* connections by the period to start.

So, what I'll probably do is make it configurable, as a fraction of the bucket fill rate.